### PR TITLE
Standardize metrics help message

### LINKED ISSUE #3981: Standardize metrics help message
**Task description**:
This task aims to standardize some metrics help messages. 
See https://github.com/karmada-io/karmada/pull/3949#discussion_r1301072318 for more details.

**Solution**:
This is the proposed changes:
```diff
diff --git a/pkg/metrics/resource.go b/pkg/metrics/resource.go                                                 
index 71e96d871..3555f0194 100644                                                                              
--- a/pkg/metrics/resource.go                                                                                  
+++ b/pkg/metrics/resource.go                                                                                  
@@ -22,24 +22,24 @@ const (                                                                                    
 var (                                                                                                         
        findMatchedPolicyDurationHistogram = prometheus.NewHistogramVec(prometheus.HistogramOpts{              
                Name:    resourceMatchPolicyDurationMetricsName,                                               
-               Help:    "Duration in seconds to find a matched propagation policy for the resource template.",
+               Help:    "Duration in seconds to find a matched PropagationPolicy or ClusterPropagationPolicy for the resource templates.",
                Buckets: prometheus.ExponentialBuckets(0.001, 2, 12),
        }, []string{})

        applyPolicyDurationHistogram = prometheus.NewHistogramVec(prometheus.HistogramOpts{
                Name:    resourceApplyPolicyDurationMetricsName,
-               Help:    "Duration in seconds to apply a propagation policy for the resource template. By the result, 'error' means a resource template failed to apply the policy. Otherwise 'success'.",
+               Help:    "Duration in seconds to apply a PropagationPolicy or ClusterPropagationPolicy for the resource templates. By the result, 'error' means a resource template failed to apply the policy. Otherwise 'success'.",
                Buckets: prometheus.ExponentialBuckets(0.001, 2, 12),
        }, []string{"result"})

        policyApplyAttempts = prometheus.NewCounterVec(prometheus.CounterOpts{
                Name: policyApplyAttemptsMetricsName,
-               Help: "Number of attempts to be applied for a propagation policy. By the result, 'error' means a resource template failed to apply the policy. Otherwise 'success'.",
+               Help: "Number of attempts to be applied for a PropagationPolicy or ClusterPropagationPolicy. By the result, 'error' means a resource template failed to apply the policy. Otherwise 'success'.",
        }, []string{"result"})

        syncWorkDurationHistogram = prometheus.NewHistogramVec(prometheus.HistogramOpts{
                Name:    syncWorkDurationMetricsName,
-               Help:    "Duration in seconds to sync works for a binding object. By the result, 'error' means a binding failed to sync works. Otherwise 'success'.",
+               Help:    "Duration in seconds to sync works for ResourceBinding and ClusterResourceBinding objects. By the result, 'error' means a binding failed to sync works. Otherwise 'success'.",
                Buckets: prometheus.ExponentialBuckets(0.001, 2, 12),
        }, []string{"result"})

```

**Who can join or take the task**:

The good first issue is intended for `first-time contributors` to get started on his/her contributor journey.

After a contributor has successfully completed 1-2 good first issue's,
they should be ready to move on to `help wanted` items, saving the remaining `good first issue` for other new contributors.

**How to join or take the task**:

Just reply on the issue with the message `/assign` in a separate line.

Then, the issue will be assigned to you.

**How to ask for help**:

If you need help or have questions, please feel free to ask on this issue.
The issue author or other members of the community will guide you through the contribution process.

### PR DESCRIPTION
**Pull request by HeavenTonight**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #3981 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve
/kind cleanup

Thanks @HeavenTonight for the help.

**Comment by RainbowMango** [comment]
Hi @HeavenTonight Can you help to fix the failing unit tests as well? Sorry, I didn't notice these tests.
```
--- FAIL: TestIncomingBindingMetrics (0.01s)
    --- FAIL: TestIncomingBindingMetrics/add_resourceBinding (0.00s)
    --- FAIL: TestIncomingBindingMetrics/update_resourceBinding (0.00s)
    --- FAIL: TestIncomingBindingMetrics/policy_associated_with_resourceBinding_changed (0.00s)
    --- FAIL: TestIncomingBindingMetrics/policy_associated_with_clusterResourceBinding_changed (0.00s)
    --- FAIL: TestIncomingBindingMetrics/cluster_which_matches_the_existing_propagationPolicy_changed (0.00s)
    --- FAIL: TestIncomingBindingMetrics/cluster_which_matches_the_existing_clusterPropagationPolicy_changed (0.00s)
    --- PASS: TestIncomingBindingMetrics/schedule_attempt_success (0.00s)
    --- FAIL: TestIncomingBindingMetrics/schedule_attempt_failure (0.00s)
```
You can run these tests under the root of `karmada` repo by command: `# go test ./pkg/scheduler/... -run=TestIncomingBindingMetrics`.

**Comment by HeavenTonight** [comment]
> Hi @HeavenTonight Can you help to fix the failing unit tests as well? Sorry, I didn't notice these tests.
> 
> 嗨@HeavenTonight你能帮忙修复失败的单元测试吗？抱歉，我没有注意到这些测试。
> 
> ```
> --- FAIL: TestIncomingBindingMetrics (0.01s)
>     --- FAIL: TestIncomingBindingMetrics/add_resourceBinding (0.00s)
>     --- FAIL: TestIncomingBindingMetrics/update_resourceBinding (0.00s)
>     --- FAIL: TestIncomingBindingMetrics/policy_associated_with_resourceBinding_changed (0.00s)
>     --- FAIL: TestIncomingBindingMetrics/policy_associated_with_clusterResourceBinding_changed (0.00s)
>     --- FAIL: TestIncomingBindingMetrics/cluster_which_matches_the_existing_propagationPolicy_changed (0.00s)
>     --- FAIL: TestIncomingBindingMetrics/cluster_which_matches_the_existing_clusterPropagationPolicy_changed (0.00s)
>     --- PASS: TestIncomingBindingMetrics/schedule_attempt_success (0.00s)
>     --- FAIL: TestIncomingBindingMetrics/schedule_attempt_failure (0.00s)
> ```
> 
> You can run these tests under the root of `karmada` repo by command: `# go test ./pkg/scheduler/... -run=TestIncomingBindingMetrics`.
> 
> 您可以通过命令在 karmada repo 的根目录下运行这些测试： # go test ./pkg/scheduler/... -run=TestIncomingBindingMetrics。

yes,i will fix it

**Comment by HeavenTonight** [comment]
> Hi @HeavenTonight Can you help to fix the failing unit tests as well? Sorry, I didn't notice these tests.↳
> 
> ```
> --- FAIL: TestIncomingBindingMetrics (0.01s)
>     --- FAIL: TestIncomingBindingMetrics/add_resourceBinding (0.00s)
>     --- FAIL: TestIncomingBindingMetrics/update_resourceBinding (0.00s)
>     --- FAIL: TestIncomingBindingMetrics/policy_associated_with_resourceBinding_changed (0.00s)
>     --- FAIL: TestIncomingBindingMetrics/policy_associated_with_clusterResourceBinding_changed (0.00s)
>     --- FAIL: TestIncomingBindingMetrics/cluster_which_matches_the_existing_propagationPolicy_changed (0.00s)
>     --- FAIL: TestIncomingBindingMetrics/cluster_which_matches_the_existing_clusterPropagationPolicy_changed (0.00s)
>     --- PASS: TestIncomingBindingMetrics/schedule_attempt_success (0.00s)
>     --- FAIL: TestIncomingBindingMetrics/schedule_attempt_failure (0.00s)
> ```
> 
> You can run these tests under the root of `karmada` repo by command: `# go test ./pkg/scheduler/... -run=TestIncomingBindingMetrics`.↳

yes, i will fix it

**Comment by HeavenTonight** [comment]
I am very confused,why the error in ```TestIncomingBindingMetrics``` unit test?:sweat_smile:
image
@RainbowMango Can you provide me with some help? thanks

**Comment by RainbowMango** [comment]
OK. I'm looking at it.

File: pkg/scheduler/scheduler_metrics_test.go [hunk 1]
```diff
@@ -100,7 +100,8 @@ func TestIncomingBindingMetrics(t *testing.T) {
 			crbInvolved: false,
 			trigger:     addBinding,
 			want: `
-			# HELP karmada_scheduler_queue_incoming_bindings_total Number of bindings added to scheduling queues by event type.
+			# HELP karmada_scheduler_queue_incoming_bindings_total Number of ResourceBinding and ClusterResourceBinding objects ad
+ded to scheduling queues by event type.
```
Comment (RainbowMango) [inline pkg/scheduler/scheduler_metrics_test.go:104]:
```suggestion
			# HELP karmada_scheduler_queue_incoming_bindings_total Number of ResourceBinding and ClusterResourceBinding objects added to scheduling queues by event type.
```
Same as below.

Comment (HeavenTonight) [reply pkg/scheduler/scheduler_metrics_test.go:104]: 😅 Sorry,this is stupid mistake.

**Pull request closed without merging**